tests/common/dualtor/dualtor_utils.py

```diff
@@ -0,0 +1,115 @@
+import logging
+import pytest
+import json
+import urllib2
+from tests.common.helpers.assertions import pytest_assert
+
+logger = logging.getLogger(__name__)
+
+TOR_A = "tor_a"
+TOR_B = "tor_b"
+NIC = "nic"
+
+DROP = "drop"
+OUTPUT = "output"
+# Here we assume that the group name of host server starts with 'vm_host_'.
+VMHOST_PREFIX = "vm_host_"
+VMHOST_ADDRESS = ""
+
+@pytest.fixture(scope='session', autouse=True)
+def mux_simulator_server_address(tbinfo, localhost, duthost):
+    """
+    A session level fixture to retrieve the address of mux simulator address
+    Args:
+        tbinfo: A session level fixture
+        localhost: A session level fixture
+    Returns:
+        str: The address (including vmset path) of mux simulator server, like http://10.0.0.64:8080/mux/vms17-8
+    """
+    vmhost_server_name = tbinfo['server']
+    vmset_name = tbinfo['group-name']
+    # We assume that if server name in testbed is server_#, then the vmhost server name is vm_host_#
+    vmhost_group_name = VMHOST_PREFIX + vmhost_server_name.split('_')[-1]
+    inv_mgr = localhost.host.options['inventory_manager']
+    all_hosts = inv_mgr.get_hosts(pattern=vmhost_group_name)
+    assert len(all_hosts) == 1
+    vmhost_server = inv_mgr.get_host(all_hosts[0].get_name()).vars['ansible_host']
+    vmhost_port = duthost.host.options["variable_manager"]._hostvars[all_hosts[0].get_name()]['mux_simulator_port']
+    global VMHOST_ADDRESS
```

Comment:
I feel that using a global variable to return value is not a good way. Using an 'autouse' fixture to return the mux simulator URL is not a good way either.

Reply:
Updated. Thanks for your great help.